test/infra/async_utils.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (c) Microsoft Corporation.
```

Comment:
Surprisingly enough, nothing I could find.

https://github.com/groove-x/trio-util has `wait_any` which is the same as `race_tasks`, but this is based around trio rather than asyncio. There's also a [lengthy discussion about it on the trio issue tracker](https://github.com/python-trio/trio/issues/1089).

I couldn't find any precedent for EventLoopThread.